src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -403,6 +414,12 @@ export async function makePythonCodeLensProvider(
             document: vscode.TextDocument,
             token: vscode.CancellationToken
         ): Promise<vscode.CodeLens[]> => {
+            // Try to activate the Python Extension before requesting symbols from a python file
+            await activatePythonExtensionIfInstalled()
+            if (token.isCancellationRequested) {
```

Comment:
Is there any better way to use cancellation tokens? this looks like it will become a very painful pattern in vscode

Reply:
Cancellation tokens have an `onCancellationRequested` event which is useful in some situations. In this case, I don't know if we could use it to stop the code flow

Reply:
In case you're wondering, the `[]` is returned based on the documentation:

> An array of code lenses or a thenable that resolves to such. The lack of a result can be signaled by returning undefined, null, or an empty array.